Add enhancements to module VMCPFunctions

equelin opened the pull request:
Allow more flexibility for providing clusters to the functions and add support to `-Whatif` and `-Confirm` parameters.

New usage examples:

``` Powershell
    Get-VMCPSettings
```

This will show you the VMCP settings for all the clusters

``` Powershell
    Get-VMCPSettings -Cluster 'CLUSTER01','CLUSTER02'
```

This will show you the VMCP settings for clusters CLUSTER01 and CLUSTER02

``` Powershell
    Set-VMCPSettings -enableVMCP:$False -VmStorageProtectionForPDL `
    disabled -VmStorageProtectionForAPD disabled `
    -VmTerminateDelayForAPDSec 600 -VmReactionOnAPDCleared none 
```

This will disable VMCP and configure the Settings on all clusters available

Tested on:
- VMware vSphere PowerCLI 6.3 Release 1 build 3737840
- Powershell 5
- Windows 10

kmruddy commented:
get-vmcpsetting:  approved from my perspective, works as intended in a nicer way than the previous

set-vmcpsetting: has some work that’s needed
Not sure if it’s missing the input or it’s intentional, but when invoking a change I’d like to not see it done against a group of objects in a generic sense. 

Example: Line 175, it will apply the change of VMCP settings to all clusters available in the current session

Some work needs done to the enableVMCP parameter (although it looks like this existed even in the original commit), it doesn’t take “true” or “false” and if you run it without specifying the parameter it won’t take any of the input when prompted. Error indicates an issue converting a string to a boolean. 

I didn’t go too much further than that…

equelin commented:
Hello @kmruddy and thank you for the review!

Line 175: The user has to provide one cluster (or more) to reconfigure. 
Line 180: The `enableVMCP` parameters is now a boolean
Line 314 to 318: I've added a `Wait-Task` command to be sure that the cluster has been reconfigured before showing up the result

Also I've modified more deeply Set-VMCPsetting, it's possible now to change only the needed parameters without providing the others. It's a big change in the philisophy of the function (IMHO it adds more granularity) so it would be nice that @vtagion provides is point of view.

kmruddy commented:
vSphere: 6.0 & 6.5
PowerCLI: 6.5 R1

Everything is looking good!

Approved.

kmruddy commented:
Due to an update in the approval policy (https://github.com/vmware/PowerCLI-Example-Scripts#approval-of-additions), this pull request has been committed. Please use this repo’s Issues section for any additional comments.

Thanks!
